### PR TITLE
Check on platform row

### PR DESCRIPTION
**Pull request by shenbakavanee-deriv**
In this test case we will be checking the platforms under [User-friendly trading platforms] in Deriv.com homepage and check the image & descriptions for ROW region.

Card: https://app.clickup.com/t/20696747/DPROD-3900

### CODE REVIEW

File: cypress/e2e/wip/platform-row.cy.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import '@testing-library/cypress/add-commands'
+
+describe('QATEST-1350 - Our Platform ROW', () => 
+{
+    it('should be able to navigate to Dbot page from home page and validate the page content and links in Desktop for ROW', () => {
+        cy.c_visitResponsive('', {size:'desktop'})
```
Comment (meeyun-deriv) [inline cypress/e2e/wip/platform-row.cy.js:6]: please include mobile test as well

File: cypress/e2e/wip/platform-row.cy.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        cy.findByText("User-friendly trading platforms, on any device").click()
+        cy.findByText("All").click();
+        
+        // Check CFDs tab
+        cy.get('#platformtab-trigger-1').click();
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/platform-row.cy.js:11]: Please use:
```suggestion
        cy.findByRole('tab', { name: 'CFDs' })
```

Comment (meeyun-deriv) [inline cypress/e2e/wip/platform-row.cy.js:11]: i believe this is clicking `Trade` from the header. If so, you can try this instead
```
homeBanner.elements.tradeMenu().click()
```

File: cypress/e2e/wip/platform-row.cy.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        // Check Deriv MT5 
+        cy.findByRole("heading", { name: "Deriv MT5" }).should('be.visible');
+        cy.get('img[alt="deriv mt5"]').should('be.visible').click();
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/platform-row.cy.js:14]: Please use:
```suggestion
        cy.findByRole('img', { name: 'deriv mt5' })
```

File: cypress/e2e/wip/platform-row.cy.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        cy.findByText("The most popular and comprehensive CFDs platform.").should('be.visible')
+        cy.findByRole('link', { name: 'Learn more About Deriv MT5' }).click();
+        cy.get('img[alt="Deriv MT5"]').first().click();
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/platform-row.cy.js:17]: Can you please use cypress testing library commands?

File: cypress/e2e/wip/platform-row.cy.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        cy.findByRole('link', { name: 'deriv logo label' }).click(); 
+        // Check Ctrader
+        cy.findByRole("heading", { name: "Deriv cTrader" }).should('be.visible');
+        cy.get('img[alt="deriv ctrader"]').should('be.visible').click();
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/platform-row.cy.js:21]: Same here

File: cypress/e2e/wip/platform-row.cy.js (hunk 3 above)
Comment (meeyun-deriv) [inline cypress/e2e/wip/platform-row.cy.js:14]:
```
cy.findAllByText('Deriv MT5').eq(0).should('be.visible').click()
```

can use this instead

File: cypress/e2e/wip/platform-row.cy.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        cy.findByText("Fast CFDs platform with inbuilt copy trading.").should('be.visible')
+        cy.findByRole('link', { name: 'Learn more About Deriv cTrader' }).click();
+        cy.findByRole('img', { name: '_t_Deriv ctrader logo_t_' }).click();
+        cy.findByRole('link', { name: 'deriv logo label' }).click(); 
+        //Check Deriv X
+        cy.findByRole("heading", { name: "Deriv X" }).should('be.visible');
+        cy.get('img[alt="deriv x"]').should('be.visible').click();
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/platform-row.cy.js:28]: Same

File: cypress/e2e/wip/platform-row.cy.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        cy.findByText("User-friendly and customisable CFDs platform.").should('be.visible')
+        cy.findByRole('link', { name: 'Learn more About Deriv X' }).click();
+        cy.get('img[alt="Deriv X"]').first().click();
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/platform-row.cy.js:31]: Here as well

**Review by abuzar-deriv — CHANGES_REQUESTED** [review]
Looks fine in general but can you please replace all instances of `cy.get` with cypress testing library commands? You may use playwright code gen to find the locators.

File: cypress/e2e/wip/platform-row.cy.js (hunk 4 above)
Comment (meeyun-deriv) [inline cypress/e2e/wip/platform-row.cy.js:17]: do use testing library queries for all. If it is not possible due to the current code, we need to raise a card for FE developer to update that instead

File: cypress/e2e/wip/platform-row.cy.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        cy.findByRole('link', { name: 'deriv logo label' }).click(); 
+
+        // Check Bot tab
+        cy.get('#platformtab-trigger-3').click();
+        cy.findByRole("heading", { name: "Deriv Bot" }).should('be.visible');
+        cy.get('img[alt="deriv bot"]').should('be.visible').click();
+        cy.findByText("Automated trading. No coding required.").should('be.visible')
+        cy.findByRole('link', { name: 'Learn more About Deriv Bot' }).click();
+        cy.findByRole('img', { name: 'Deriv Bot' }).click();
+        cy.findByRole('link', { name: 'deriv logo label' }).click();
```
Comment (meeyun-deriv) [inline cypress/e2e/wip/platform-row.cy.js:68]: for the validation step, we should use `....should('something')...` instead of an action

**Review by meeyun-deriv — CHANGES_REQUESTED** [review]

**Review by meeyun-deriv — CHANGES_REQUESTED** [review]
move this test from `wip` folder to `full` folder

**Review by meeyun-deriv — APPROVED** [review]
LGTM

**Pull request merged**